scripts/workspace_cromwell.py

```diff
@@ -0,0 +1,295 @@
+import os
+import json
+from pathlib import Path
+import requests
+import subprocess
+import sys
+import time
+import uuid
+
+
+# If a CROMWELL app exists return (app_name, status), otherwise (None, None).
+def check_for_app(env):
+    list_apps_url = f'{env["leonardo_url"]}/api/google/v1/apps/{env["google_project"]}'
+    r = requests.get(
+        list_apps_url,
+        params={
+          'includeDeleted': 'false'
+        },
+        headers = {
+            'Authorization': f'Bearer {env["token"]}',
+            'Referer': 'https://bvdp-saturn-dev.appspot.com/'
+        }
+    )
+    r.raise_for_status()
+
+    for potential_app in r.json():
+        if potential_app['appType'] == 'CROMWELL':
+            potential_app_name = potential_app['appName']
+            potential_app_status = potential_app['status']
+
+            # We found a CROMWELL app in the correct google project and owned by the user. Now just check the workspace:
+            _, workspace_namespace, workspace_name, _ = get_app_details(env, potential_app_name)
+            if workspace_namespace == env['workspace_namespace'] and workspace_name == env['workspace_name']:
```

Comment:
Should we also check that the app was created by the same user? Because if I am understanding correctly, multiple users can each create Cromwell app in the same workspace and if the current user is owner/writer in workspace they might have the permissions to see other apps in the list apps api? And if that is the case we might wrongly delete app created by another user?

Reply:
Hmm, good question. I don't think this would be a problem because the current model only shows apps created by you (since there are no multi-tenant apps right now).

I will test this out to validate it.